[Wang Mo] iP

WM71811 opened the pull request.

Commit:
Add Gradle support

WM71811 committed:
implement level 1

WM71811 committed:
implement level 2

WM71811 committed:
implement Task class and TaskSimulator class

WM71811 committed:
implement level 3

WM71811 committed:
implement level 3 (modified)

WM71811 committed:
implement A-inheritance

WM71811 committed:
implement level 4

WM71811 committed:
implement A-TextUiTesting

WM71811 committed:
implement A-Exceptions

WM71811 committed:
implement level 5

WM71811 committed:
implement level 6

WM71811 committed:
update A-TextUiTesting

Include testing for delete and error-handling(Level 5 and Level 6)

WM71811 committed:
Update to comply with the coding standard

WM71811 committed:
Implement level-7

The past tasks added or deleted are not saved in a file before.

It should be saved in a file for retrieval and the file should be updated after each operation, so that data would be remembered after the operations.

Let's create a Storage class to store the data related to tasks and enable write to the file and read data from the file.

This is done so that the user could have easy reference to tasks they added before.

WM71811 committed:
Implement level-8

WM71811 committed:
Merge branch 'branch-Level-8'

WM71811 committed:
Implement A-MoreOOP

The classes are not yet refactored using more OOP concepts.

This need to change since such code is harder to maintain.

Let's extract the classes Ui, Storage, Parser, TaskList and Command classes.

This will make the logic looks more clear by using more OOP concepts.

WM71811 committed:
Implement A-Packages

WM71811 committed:
Add Junit tests

Tests are not present to test the behavior of the code.

Junit tests should be added to test the behavior of the code since it helps to spot errors.

Let's add Junit tests.

This allows the testing of methods become facilitated.

WM71811 committed:
Implement A-Jar

WM71811 committed:
Add Javadoc documents

Currently there is no javadoc written.

Javadoc is important as it helps to explain the usages of each method and class.

Let's write Javadoc documents for more than half of the public methods/classes.

WM71811 committed:
Implement A-CodingStandard

WM71811 committed:
Implement Level-9

WM71811 committed:
Merge branch 'branch-A-CodingStandard'

WM71811 committed:
Merge branch 'branch-Level-9'

Commit:
build.gradle: Update version to 8.29

WM71811 committed:
Update Javadoc

WM71811 committed:
Merge branch 'add-gradle-support' into A-Gradle

WM71811 committed:
Implement A-Gradle

WM71811 committed:
Add gradle doc

WM71811 committed:
Merge branch 'A-Gradle'

WM71811 committed:
Revert "Merge branch 'A-Gradle'"

This reverts commit 65e7b9017bd0865bbce9e0f07ab670c994c6789a, reversing
changes made to 25c2207dd5709c4682c43698b5c49a53cb979209.

WM71811 committed:
Rename the branch to branch-A-Gradle

WM71811 committed:
Merge branch 'branch-A-Gradle'

WM71811 committed:
Revert "Revert "Merge branch 'A-Gradle'""

This reverts commit f4b89d4c4bff96a23fa355c2c8eaa4bef893d1bb.

DreamerDragon commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -0,0 +1,52 @@
+ * Represents a DeleteCommand and handles methods related to commands
+ * about deleting a task.
+ */
+public class DeleteCommand extends Command {
```
Perhaps the class name could change to CommandDeleter or other nouns? Reference: Use nouns for classes/variables and verbs for methods/functions.

DreamerDragon replied:
Ohh I think I misinterpret this as a verb at first, but from the rest of the class names,  I could tell this is a noun as well

WM71811 replied:
Okay thanks!

DreamerDragon commented on src/main/java/duke/command/ErrorCommand.java:
```diff
@@ -0,0 +1,47 @@
+ * Represents an ErrorCommand and handles methods related to
+ * sending error message to users.
+ */
+public class ErrorCommand extends Command {
```
I like how you name your command classes which is pretty clear for me to follow!!

WM71811 replied:
Thank you!

DreamerDragon commented on src/main/java/duke/exception/InvalidDeadlineException.java:
```diff
@@ -0,0 +1,17 @@
+ * Represents an InvalidDeadlineException and handles the exceptions related
+ * to invalid deadline tasks.
+ */
+public class InvalidDeadlineException extends InvalidInputException {
```
I like how you clearly named each of the exceptions from which I can instantly get the meaning!

WM71811 replied:
Thanks!

DreamerDragon commented on src/main/java/duke/parser/TimeParser.java:
```diff
@@ -0,0 +1,42 @@
+ * Represents a TimeParser and consists of methods related to
+ * parsing the string representing the time of tasks.
+ */
+public class TimeParser {
```
I like the way that you name it as a noun instead of ParseTime!

WM71811 replied:
Thank you!

DreamerDragon reviewed (commented):
Overall your naming are clear, but maybe some class names at first is a bit confusing for me to find out whether it is a noun or verb. I learnt from your codes and coding standards!! Well Done Wang Mo!

WM71811 committed:
implement GUI

The program does not have GUI yet.

GUI is required to have a good user interface.

Let's implement GUI.

A GUI implemented can allow users to use the program easily.

WM71811 committed:
Merge branch 'branch-Level-10'

WM71811 committed:
Update javadoc

moyj01 commented on src/main/java/duke/command/AddCommand.java:
```diff
@@ -0,0 +1,50 @@
+package duke.command;
+import duke.storage.Storage;
+import duke.task.TaskList;
+import duke.ui.Ui;
+import duke.exception.InvalidInputException;
+import duke.task.Task;
```
Hi Wang Mo, I noticed that some imports are not arranged in lexicographical order. For example, 'duke.exception.InvalidInputException' import could be placed before 'duke.ui.Ui' and 'duke.task.Task' import could be placed before 'duke.ui.Ui'. This would make the ordering of import statements more consistent.

moyj01 replied:
To add on, I like the clear and concise Javadocs you have written. Your code is enjoyable to read as well 👍 .

WM71811 replied:
Yes, I am working on that. Thank you so much!

moyj01 commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,86 @@
+     * @throws InvalidRequestException If the command is invalid.
+     */
+    public void deleteTask(int index) throws InvalidRequestException {
+            if (listOfTasks.size() < index || index < 0) {
```
There are some issues with the indentation in this class. For example,  the expected number of spaces should be 8 instead of 12 in line 40.

WM71811 replied:
Yes, I have changed that. Thank you for pointing that out!

moyj01 commented on src/main/java/duke/exception/InvalidRequestException.java:
```diff
@@ -0,0 +1,19 @@
+package duke.exception;
+
+import duke.exception.InvalidInputException;
```
It appears to me that this import was not used. Maybe you could take a look and check if it could be removed?

WM71811 replied:
Thank you for pointing it out! I have deleted the unused import statement.

moyj01 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,80 @@
+            fxmlLoader.setRoot(this);
+            fxmlLoader.load();
+            dialog.setText(input);
+            Circle clip = new Circle(displayPicture.getFitWidth()/2,
+                    displayPicture.getFitHeight()/2, displayPicture.getFitWidth()/2);
```
I noticed that the '/' was not preceded and followed by whitespace in this class. Great job on ensuring that operators are surrounded by whitespace in other classes 👍

WM71811 replied:
Thank you for pointing it out! I have changed it to follow the coding standard.

WM71811 committed:
Update javadocs and GUI

WM71811 committed:
Update to follow CheckStyle

WM71811 committed:
Merge branch 'branch-A-CheckStyle'

WM71811 committed:
Update minor changes

WM71811 committed:
Implement A-Assertions

WM71811 committed:
Implement A-CodeQuality

WM71811 committed:
Merge pull request #1 from WM71811/branch-A-Assertions

Implement A-Assertions

WM71811 committed:
Merge branch 'master' into branch-A-CodeQuality

WM71811 committed:
Merge pull request #2 from WM71811/branch-A-CodeQuality

Implement A-CodeQuality

WM71811 committed:
Implement C-DetectDuplicates

WM71811 committed:
Merge pull request #3 from WM71811/branch-C-DetectDuplicates

Implement C-DetectDuplicates

WM71811 committed:
Implement A-BetterGui

WM71811 committed:
Edit README

WM71811 committed:
Edit user guide

WM71811 committed:
Set theme jekyll-theme-cayman

WM71811 committed:
Update testing

WM71811 committed:
Update README.md

WM71811 committed:
Update README.md

WM71811 committed:
Edit readme

WM71811 committed:
Render github page

WM71811 committed:
Update exceptions

WM71811 committed:
Edit readme and add help command

WM71811 committed:
Update README.md

WM71811 committed:
Edit readme

WM71811 committed:
Improve code quality
